### ISSUE TITLE
AVX2 detection on Ryzen 7 CPU

### ISSUE THREAD

**Issue by atomsymbol** [ValveSoftware/steam-for-linux#6561]
#### Your system information

* Steam client version (build number or date): 2019-sep-30 21:41:49
* Distribution (e.g. Ubuntu): Gentoo
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:

Steam - System Information isn't detecting AVX2 support on an AMD Ryzen 7 CPU. Please see the second-to-last line in the content area of the attached screenshot.

/proc/cpuinfo contains sha_ni but Steam system information reports SHA is unsupported.

Hard disk space space is being reported in Mb (megabits) instead of MB or MiB (megabytes).

[sysinfo 2019-10-06.txt](https://github.com/ValveSoftware/steam-for-linux/files/3694787/sysinfo.2019-10-06.txt)

![Screenshot_20191006_212722](https://user-images.githubusercontent.com/22368256/66274588-65082000-e880-11e9-8f88-245a5191dfee.png)

**Comment by Leopard1907** [comment]
It is same for everybody i think. It says my i7 7700HQ doesn't support AVX2 either , which is not the case.

https://ark.intel.com/content/www/us/en/ark/products/97185/intel-core-i7-7700hq-processor-6m-cache-up-to-3-80-ghz.html

**Comment by suokko** [comment]
Seems like AVX2 detection is broken. I noticed same issue with Intel i5-5200U (Broadwell). I know it has AVX2 because I have used AVX2 instructions on my own code.

I don't know if sha detection is broken too as my cpu doesn't support that extension.

**Comment by atomsymbol** [comment]
An update based on [Steam HW survey](https://store.steampowered.com/hwsurvey?platform=linux) for January 2020:

AVX2: 0.01% (-0.04% relative to December 2019)

With Steam for Linux unable to detect AVX2 at all, it is curious that the reported number is above zero percent.

**Comment by kode54** [comment]
It does appear they are detecting SSE4a properly, though. Considering that unlike current Intel processors, AMD decided to maintain support for that in their current processors. Intel decided instead to drop it when they added SSE4.1, or possibly 4.2. The SSE4a stat is within a percent or two of the AuthenticAMD statistic.

**Comment by atomsymbol** [comment]
Indeed, SSE4A percentage is correct. Intel's Instruction Set Manual doesn't mention any SSE4A, because no Intel CPU ever supported those instructions.

**Comment by atomsymbol** [comment]
@kisak-valve It is good that this issue is being tracked on GitHub, but is there any kind of estimate when it might get fixed? Why keep reporting in the [hwsurvey](https://store.steampowered.com/hwsurvey?platform=linux), again and again every month, something that isn't being accurately measured?

**Comment by atomsymbol** [comment]
An update based on [Steam HW survey](https://store.steampowered.com/hwsurvey?platform=linux) for May 2020:

AVX2: 0.03% (-0.02% relative to April 2020)

With Steam for Linux unable to detect AVX2 at all, it is curious that the reported number is above zero percent.

**Comment by landjalan** [comment]
one year later, the steam survey still says AVX2: unsupported as well as AVX512x: unsupported. in make.conf i have CPU_FLAGS_x86="xxx avx2 xxx ....". current kernel 5.7.2

**Comment by atomsymbol** [comment]
> one year later, the steam survey still says AVX2: unsupported as well as AVX512x: unsupported. in make.conf i have CPU_FLAGS_x86="xxx avx2 xxx ....". current kernel 5.7.2

It appears this issue will be resolved only if Valve receives a court order to fix it because of _regularly knowingly publishing_ misleading CPU market share data.

**Comment by condekind** [comment]
+1, still a problem for Ryzen 3000 series.
AMD 3900X, Arch Linux, kernel 5.8.10, no DE (i3wm).

**Comment by baryluk** [comment]
Same here (ThreadRipper 2950X). Not only `AVX2` is not detected. But also `SHA` is not detected.

```
# inxi -C -f
CPU:       Info: 16-Core (2-Die) model: AMD Ryzen Threadripper 2950X bits: 64 type: MT MCP MCM L2 cache: 8192 KiB 
           Speed: 1994 MHz min/max: 2200/3500 MHz Core speeds (MHz): 1: 2004 2: 2193 3: 1893 4: 1876 5: 2019 6: 2157 7: 1888 
           8: 2016 9: 1974 10: 2003 11: 2116 12: 1893 13: 1986 14: 1900 15: 2026 16: 1908 17: 2149 18: 1895 19: 1971 20: 1976 
           21: 1977 22: 2039 23: 1919 24: 1937 25: 1974 26: 1892 27: 2034 28: 2027 29: 1893 30: 2001 31: 2037 32: 1893 
           Flags: 3dnowprefetch abm adx aes amd_dcm aperfmperf apic arat avic avx avx2 bmi1 bmi2 bpext clflush clflushopt 
           clzero cmov cmp_legacy constant_tsc cpb cpuid cr8_legacy cx16 cx8 de decodeassists extapic extd_apicid f16c 
           flushbyasid fma fpu fsgsbase fxsr fxsr_opt ht hw_pstate ibpb irperf lahf_lm lbrv lm mca mce misalignsse mmx mmxext 
           monitor movbe msr mtrr mwaitx nonstop_tsc nopl npt nrip_save nx osvw overflow_recov pae pat pausefilter pclmulqdq 
           pdpe1gb perfctr_core perfctr_llc perfctr_nb pfthreshold pge pni popcnt pse pse36 rdrand rdseed rdtscp rep_good sep 
           sev sha_ni skinit smap smca sme smep ssbd sse sse2 sse4_1 sse4_2 sse4a ssse3 succor svm svm_lock syscall tce 
           topoext tsc tsc_scale v_vmsave_vmload vgif vmcb_clean vme vmmcall wdt xgetbv1 xsave xsavec xsaveerptr xsaveopt 
           xsaves 
```


![steam-hw-dialog](https://user-images.githubusercontent.com/110322/94876400-18c9ce80-0447-11eb-9f40-c2b2248c40cd.png)

**Comment by mradalbert** [comment]
Same thing on i7-4720HQ.
Manjaro, everything up to date.
```
 [wojtek@wojtek-portable ~]$ inxi -C -f
 CPU:       Info: Quad Core model: Intel Core i7-4720HQ bits: 64 type: MT MCP L2 cache: 6 MiB 
            Speed: 3418 MHz min/max: 800/3600 MHz Core speeds (MHz): 1: 3418 2: 3345 3: 3354 4: 3428 
            5: 3501 6: 3316 7: 3403 8: 3201 
            Flags: abm acpi aes aperfmperf apic arat arch_perfmon avx avx2 bmi1 bmi2 bts clflush 
            cmov constant_tsc cpuid cpuid_fault cx16 cx8 de ds_cpl dtes64 dtherm dts epb ept ept_ad 
            erms est f16c flexpriority fma fpu fsgsbase fxsr ht ida invpcid invpcid_single lahf_lm 
            lm mca mce mmx monitor movbe msr mtrr nonstop_tsc nopl nx pae pat pbe pcid pclmulqdq 
            pdcm pdpe1gb pebs pge pln pni popcnt pse pse36 pti pts rdrand rdtscp rep_good sdbg sep 
            smep ss sse sse2 sse4_1 sse4_2 ssse3 syscall tm tm2 tpr_shadow tsc tsc_adjust vme vmx 
            vnmi vpid x2apic xsave xsaveopt xtopology xtpr
```
![obraz](https://user-images.githubusercontent.com/26419317/106368315-fab6a100-6348-11eb-803b-c06b016cf0d1.png)

**Comment by rezad1393** [comment]
same here with ryzen 7 5700u vega 8 .
archlinux 5.14.3-arch1-1
